### ISSUE TITLE
s3 backend ignores key prefixes

### ISSUE THREAD

**Issue by eropple** [pulumi/pulumi#2812]
As per conversation with @stack72 --

It looks like `pulumi login s3://some-bucket/some-key` doesn't act as expected. It appears to be chopping the key portion off of the `s3` URL. Given the above URL, I expect my stacks to be stored at `s3://some-bucket/some-key/.pulumi`; instead they're stored at `s3://some-bucket/.pulumi` (which has actually caused me some problems today, in that I've had to go redo the way we handle micro-stacks).

Example in context:

```
$ pulumi login s3://hydrow-pulumi/foobar
Logged into err-laptop.local as ed (s3://hydrow-pulumi/foobar)
$ pulumi whoami --verbose
User: ed
Backend URL: s3://hydrow-pulumi/foobar
```

So my guess is that it's being chopped somewhere along the line.

**Comment by ellismg** [comment]
This was a problem with the upstream library not respecting these prefixes (in a by design manner that we did not understand).  Should be fixed with #2797.  Note, once you adopt a CLI with these changes, you'll have to ensure the `.pulumi` folder is in the correct location, as implied by the URL.

**Comment by ellismg** [comment]
Fixed by https://github.com/pulumi/pulumi/pull/2797
